src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -11638,6 +11638,77 @@ .maxstack  2
             }
         }
 
+        [Fact]
+        public void SpanArgument_Constructor_02()
+        {
+            string source = """
+                using System;
+                record class A<T>(T[] F)
+                {
+                    public static T[] ToArray(ReadOnlySpan<T> s) => s.ToArray();
+                }
+                record class B<T>(T x, T y, T z) : A<T>(ToArray([x, y, z]));
```

Comment:
nit: it feels like the test is not verifying the value of what is actually passed to the base constructor.

Reply:
nevermind, we can trust ToArray to give an equivalent collection as the span. I just didn't have my records brain switched on when I reviewd this.